Update README to provide more clear examples

alissa-huskey opened the pull request:
I updated the `README` with some of my own code examples and did a little cleanup/polish.

* Replace "Quickstart" section with the code for the simplest possible use-case.
* Replace `WSGI` example, as I wasn't able to get the previous example working.
* Rename the previous "Quickstart" section to "How-To".
* Rename "Available funcargs" section to "Fixtures". (That seems to be the more common term. In any case, I didn't previously know that "funcargs" was another term for "fixtures", so it may be more clear to others as well.)
* Use `code` directives instead of literal blocks to syntax highlight python code.

diazona commented:
Hi @alissa-huskey, thanks for sending a contribution! I'm sorry to keep you waiting; I haven't been active on GitHub for a little while (because real life got busy) and I didn't see the notification about your PR.

If you don't mind waiting a bit longer, I'd like to come back to this and review it over the weekend when I can give it my full attention.

diazona reviewed (changes requested):
Thanks again! These changes mostly look good, I just made a few suggestions.

Sorry again for the delayed response; I'll try to keep checking back so I don't miss further activity here.

alissa-huskey commented on README.rst:
```diff
@@ -46,29 +79,39 @@ modules dealing with the actual HTTP request (of which there are more than one
 
 You simply use pytest's `funcargs feature`_ and simulate an entire server
-locally! ::
+locally! :
```
Ah, right. I missed that one when changing `::` to `.. code::`.

alissa-huskey commented:
No worries about the delay! Real life happens. I tend to be on github in fits and starts anyway, so I never expect a prompt response, if any.

I believe I addressed all of your feedback. Additionally I changed the monkeypatch link to point to the stable pytest.org docs URL.

diazona reviewed (approved):
Good call on that monkeypatch change.

Thanks again; I appreciate both the contribution and your patience!
